### PR TITLE
Browserify Crafty.js

### PR DESCRIPTION
**Pull request by kevinsimper**
Now instead of concatenate the files, Crafty.js now use the CommonJS approach with Browserify 2.

The intro.js and outro.js files is removed because they is not needed anymore.

Hashmap is exported a module.

All the test are passing.

### CODE REVIEW

File: src/2D.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
+var Crafty = require('./core.js')
+,   document = window.document
```
Comment (potomak) [inline src/2D.js]: This syntax is ugly in my opinion and more important it is inconsistent.

Comment (kevinsimper) [inline src/2D.js]: How would you suggest it looked like?

Comment (potomak) [inline src/2D.js]: I like repeating the `var` token (see http://nodeguide.com/style.html#variable-declarations), but to be consistent you should put the comma at the end of each line, see some lines below, for example at https://github.com/kevinsimper/Crafty/blob/6c8835a650b972874cc22e238f1960f16466e3c0/src/2D.js#L15-16

**Comment by potomak** [comment]
I think you can now remove also `crafty-local.js`.

File: src/2D.js (hunk 1 above)
Comment (kevinsimper) [inline src/2D.js]: I will do that when i get home :)

Comment (potomak) [inline src/2D.js]: :+1:

**Comment by sbyrnes321** [comment]
I like it! I tried it on my game and it still worked. The final crafty.js file is the same size and more-or-less equally readable.

I agree with @potomak that commas should go at the end of the line not the beginning -- jshint prefers that too.

This change would break crafty-local, so we should delete it. (Or could it be edited somehow?) The purpose of crafty-local was to allow you to change the source code after editing it, without needing to install grunt. That's nice, but we don't _really_ need it. I assume that most people who are making a javascript game would be willing to install node and grunt.

Also, this change will break the old PHP build process. So those files might as well be deleted.

**Comment by kevinsimper** [comment]
I made the chances about the commas

I also delete the php files.

**Comment by starwed** [comment]
There are still some references  to `crafty-local` in the test suite.  (In the comments, where there are instructions for running the tests without compiling.)

Presumably this new approach should make it _easier_ to load the files locally, right?

**Comment by mpetrovic** [comment]
The purpose of crafty-local was to 'build' Crafty at runtime, so work on
the engine could be done without invoking a formal build process. Just make
your changes to the files, refresh the browser and examine results.

I'm also an old fart when it comes to programming. Never touched Grunt in
my life.

**Comment by kevinsimper** [comment]
This would make it easier to load the files locally, because all the files is now valid javascript documents.

This would also make it easier to split up the code and only referencing it where it is needed.
It will also be possible now to use code from the npm community.

I have setup grunt now with the task watch, which will with the terminal command `grunt watch`, continues build the crafty.js file. So if any of the files in `src` is changed, it will build it again.

**Comment by kevinsimper** [comment]
This also makes it possible to make source maps, which will in the dev version of crafty.js map errors, directly to the individual files, like they were the 24 individual javascript files.
Then there will never be an javascript error on line 5000 anymore :)

**Comment by potomak** [comment]
This is not the purpose of this PR, but as you're changing `package.json` I think you could also move all `dependencies` to `devDependencies` as they are required to build, test, and manage Crafty code but they're not required by the Cratfy engine itself.

**Comment by kevinsimper** [comment]
You are right, that should be done.

**Comment by starwed** [comment]
I'm not so certain about moving `dependencies` to `devDependencies`.  The reason to checkout the module is to build crafty -- that's all it does!

If you disagree, let's talk about it in a separate issue.  :)

@kevinsimper As far as I can tell, this is ready to be merged.  Was there anything else you wanted to push before we do so?

**Comment by kevinsimper** [comment]
No, that would be all.

**Pull request merged**